Capstone/Feed/HomeViewController.m

```diff
@@ -0,0 +1,112 @@
+               self.colleges = [dataDictionary objectForKey:@"colleges"];
+               for (NSDictionary<NSString *, NSString *> *college in self.colleges) {
+                   [self.collegeDictionaryArrays addObject:college];
+               }
+               [self.tableView reloadData];
+            }}];
+    [task resume];
+}
+
+-(void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
+    if (searchText.length == 0) {
+        isFiltered = false;
+    } else {
+        isFiltered = true;
+        self.filteredColleges = [[NSMutableArray alloc] init];
+        for (NSDictionary<NSString *, NSString *> *college in self.colleges) {
+            NSRange collegeName = [college[@"name"] rangeOfString:searchText options:NSCaseInsensitiveSearch];
+            if (collegeName.location != NSNotFound) {
+                [self.filteredColleges addObject:college];
+            }
+        }
+    }
+    [self.tableView reloadData];
+}
+
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    if (isFiltered) {
+        return self.filteredColleges.count;
+    }
+    return self.collegeDictionaryArrays.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    HomeCell *cell = [tableView dequeueReusableCellWithIdentifier:@"HomeCell"];
+    if (isFiltered) {
+        cell.homeCollegeName.text = self.filteredColleges[indexPath.row][@"name"];
+        cell.homeCollegeDetails.text = self.filteredColleges[indexPath.row][@"shortDescription"];
+        cell.homeCollegeLocation.text = self.filteredColleges[indexPath.row][@"city"];
+        NSURL *url = [NSURL URLWithString:self.filteredColleges[indexPath.row][@"campusImage"]];
+        [cell.homeCellImage setImageWithURL:url];
+    } else {
+        cell.homeCollegeName.text = self.collegeDictionaryArrays[indexPath.row][@"name"];
+        cell.homeCollegeDetails.text = self.collegeDictionaryArrays[indexPath.row][@"shortDescription"];
+        cell.homeCollegeLocation.text = self.collegeDictionaryArrays[indexPath.row][@"city"];
+        NSURL *url = [NSURL URLWithString:self.collegeDictionaryArrays[indexPath.row][@"campusImage"]];
+        [cell.homeCellImage setImageWithURL:url];
+    }
+    return cell;
+}
+
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    UITableViewCell *cell = sender;
+    NSIndexPath *myIndexPath = [self.tableView indexPathForCell:cell];
+    NSString *nameToPass = self.collegeDictionaryArrays[myIndexPath.row][@"name"];
+    NSString *detailToPass = self.collegeDictionaryArrays[myIndexPath.row][@"shortDescription"];
+    NSString *cityToPass = self.collegeDictionaryArrays[myIndexPath.row][@"city"];
+    NSString *imageToPass = self.collegeDictionaryArrays[myIndexPath.row][@"campusImage"];
+    DetailsViewController *detailVC = [segue destinationViewController];
+    detailVC.detailName = nameToPass;
```

Comment:
in the DetailsViewController you can create a helper function set all of these values! 

`[detailVC setDetailName: nameToPass detailDetail:detailToPass detailLocation:cityToPass detailImage:imageToPass]`